example-app/App.tsx

```diff
@@ -40,6 +40,41 @@ const Header = () => {
 };
 
 const App = () => {
+  const [result, setResult] = React.useState<{
+    result: boolean | null;
+    errorReason: string | null;
+  }>({result: null, errorReason: null});
+
+  React.useEffect(() => {
+    let linking = new DeepLinking();
+    linking.addEventListener('url', async event => {
+      setResult(parseUrl(event.url));
+    });
+    async function getDeepLink() {
+      const url = await linking.getInitialUrl();
+      let res = parseUrl(url);
+      setResult(res);
+    }
+    getDeepLink();
+  }, []);
+
+  const parseUrl = (
+    url: string | undefined,
+  ): {result: boolean | null; errorReason: string | null} => {
```

Comment:
```suggestion
  ): {success: boolean | null; errorReason: string | null} => {
```

Easier to understand `result.success` instead of `result.result`?